.gitlab-ci.yml

```diff
@@ -93,26 +89,42 @@ parthenon-cuda-peformance:
       -DMPIEXEC_PREFLAGS="--allow-run-as-root"
       -DCMAKE_CXX_COMPILER=${PWD}/../external/Kokkos/bin/nvcc_wrapper
       ../ && make -j${J} && ctest -L performance
+# run regression tests with UVM until everything is on the device
+# mitigate UVM allocations on older multi-GPU systems
+    - export CUDA_LAUNCH_BLOCKING=1
+    - export CUDA_VISIBLE_DEVICES=0
+    - cd ..
+    - mkdir build-cuda-perf-uvm
+    - cd build-cuda-perf-uvm
+    - cmake -DCMAKE_BUILD_TYPE=Release -DHDF5_ROOT=/usr/local/hdf5/parallel
```

Comment:
Good idea. I think this makes sense. I'll pull the test out that covers most of the integrated features and also run it in debug mode.

Reply:
I thought about it again, and am currently favoring not to running the tests in debug mode as failover mode for now.
From my point of view, the developer will have to address issues anyway if the tests fail and to it seems easier to just create the debug build in your local development environment rather than going through a lengthy log file without any way to intervene.
Do you think regression tests with a debug build are essential at this point or would you also be happy if we postpone the introduction of those tests if this is sth that turns out to be wished for/required by many developers?

Reply:
I am of two minds on this. On the one hand, I agree with @pgrete that the developer will have to make a debug build locally. On the other hand, there have been cases we've experienced recently where a debug build failed and the release build does not. 

So I lean towards running the unit tests in debug mode *every* time but only the unit tests. Integration tests and performance tests should probably just run in release mode.

Reply:
Oh, definitely. That's exactly what I meant (we'll definitely do a debug build, but just run the unit tests in those and then for the regression tests [which take more time >few sec] we run in Release to get optimization)

Reply:
Cool.

Reply:
This is fine for now.